sycl/plugins/cuda/pi_cuda.hpp

```diff
@@ -46,29 +47,79 @@ pi_result cuda_piMemRetain(pi_mem);
 pi_result cuda_piMemRelease(pi_mem);
 pi_result cuda_piKernelRetain(pi_kernel);
 pi_result cuda_piKernelRelease(pi_kernel);
-
-
+/// \endcond
 }
 
+/// A PI platform stores all known PI devices,
+///  in the CUDA plugin this is just a vector of
+///  available devices since initialization is done
+///  when devices are used.
+///
 struct _pi_platform {
   std::vector<std::unique_ptr<_pi_device>> devices_;
 };
 
-struct _pi_device {
+/// PI device mapping to a CUdevice.
+/// Includes an observer pointer to the platform,
+/// and implements the reference counting semantics since
+/// CUDA objects are not refcounted.
+///
+class _pi_device {
   using native_type = CUdevice;
 
   native_type cuDevice_;
   std::atomic_uint32_t refCount_;
   pi_platform platform_;
 
+public:
   _pi_device(native_type cuDevice, pi_platform platform)
       : cuDevice_(cuDevice), refCount_{1}, platform_(platform) {}
 
   native_type get() const noexcept { return cuDevice_; };
 
   pi_uint32 get_reference_count() const noexcept { return refCount_; }
+
+  pi_platform get_platform() const noexcept { return platform_; };
 };
 
+/// PI context mapping to a CUDA context object.
+///
+/// There is no direct mapping between a CUDA context and a PI context,
+/// main differences described below:
+///
+/// <b> CUDA context vs PI context </b>
+///
+/// One of the main differences between the PI API and the CUDA driver API is
+/// that the second modifies the state of the threads by assigning
+/// `CUcontext` objects to threads. `CUcontext` objects store data associated
+/// with a given device and control access to said device from the user side.
+/// PI API context are objects that are passed to functions, and not bound
+/// to threads.
+/// The _pi_context object doesnt implement this behaviour, only holds the
```

Comment:
```suggestion
/// The _pi_context object doesn't implement this behavior, only holds the
```
Not sure if these are typos in British English, but other comments in these two files use `doesn't` and `behavior`.

Reply:
I should be using US English anyway, sorry